client/payment-details/payment-method/klarna/index.js

```diff
@@ -0,0 +1,146 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import PaymentDetailsPaymentMethodDetail from '../detail';
+
+/**
+ * Extracts and formats payment method details from a charge.
+ *
+ * @param {Object} charge The charge object.
+ * @return {Object}       A flat hash of all necessary values.
+ */
+const formatPaymentMethodDetails = ( charge ) => {
+	const { billing_details: billingDetails, payment_method: id } = charge;
+
+	const {
+		country: countryCode,
+		verified_name: verifiedName,
+		payment_method_category: paymentMethodCategory,
+		preferred_locale: preferredLocale,
+	} = charge.payment_method_details.klarna;
+
+	const { name, email, formatted_address: formattedAddress } = billingDetails;
+	// Use the full country name.
+	const country = wcSettings.countries[ countryCode ];
+
+	return {
+		verifiedName,
+		id,
+		name,
+		email,
+		country,
+		formattedAddress,
+		paymentMethodCategory,
+		preferredLocale,
+	};
+};
+
+/**
+ * Placeholders to display while loading.
+ */
+const paymentMethodPlaceholders = {
+	verifiedName: 'verified name placeholder',
+	fingerprint: 'fingerprint placeholder',
+	id: 'id placeholder',
+	name: 'name placeholder',
+	email: 'email placeholder',
+	formattedAddress: 'address placeholder',
+	country: 'country placeholder',
+};
+
+const KlarnaDetails = ( { charge = {}, isLoading } ) => {
+	const details = charge.payment_method_details
+		? formatPaymentMethodDetails( charge )
+		: paymentMethodPlaceholders;
+
+	const {
```

Comment:
Following my previous comment about Klarna payment method details, we could remove `verifiedName` and `country`